capabilities/writetarget/beholder/monitor/balance.go

```diff
@@ -0,0 +1,199 @@
+package monitor
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/config"
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/core"
+	"github.com/smartcontractkit/chainlink-common/pkg/utils"
+)
+
+// ChainInfo contains the chain information (used as execution context)
+type ChainInfo struct {
+	ChainFamilyName string
+	ChainID         string
+	NetworkName     string
+	NetworkNameFull string
+}
+
+// Config defines the balance monitor configuration.
+type Config struct {
+	BalancePollPeriod config.Duration
+}
+
+// BalanceClient defines the interface for getting account balances.
+type BalanceClient interface {
+	GetAccountBalance(addr string) (float64, error)
+}
+
+// BalanceMonitorOpts contains the options for creating a new balance monitor.
+type BalanceMonitorOpts struct {
+	ChainInfo           ChainInfo
+	ChainNativeCurrency string
+
+	Config           Config
+	Logger           logger.Logger
+	Keystore         core.Keystore
+	NewBalanceClient func() (BalanceClient, error)
+
+	// Maps a public key to an account address (optional, can return key as is)
+	KeyToAccountMapper func(context.Context, string) (string, error)
+}
+
+// NewBalanceMonitor returns a balance monitoring services.Service which reports the balance of all Keystore accounts.
+func NewBalanceMonitor(opts BalanceMonitorOpts) (services.Service, error) {
```

Comment:
This is something that is not directly tied to WT capability, and is a separate framework to construct a balance monitor. It should not be grouped under `/writetarget/...`

It potentially belongs in this repo, on another path.

Reply:
I think it's good enough to keep the code here in the first restructuring iteration, and move it in the next step - just please add TODO! comments and tickets to track these items.

Reply:
Added TODO [here](https://github.com/smartcontractkit/chainlink-framework/blob/0251cf4a34a7c3d8882e13d5a00a89ebe97c66ae/capabilities/writetarget/beholder/monitor/balance.go#L15)